intellij-plugin-structure/structure-hub/src/main/kotlin/com/jetbrains/plugin/structure/hub/HubPluginManager.kt

```diff
@@ -43,46 +46,82 @@ class HubPluginManager private constructor() : PluginManager<HubPlugin> {
     val tempDirectory = Files.createTempDirectory(extractDirectory, pluginFile.nameWithoutExtension).toFile()
     return try {
       pluginFile.extractTo(tempDirectory, sizeLimit)
-      loadDescriptorFromDirectory(tempDirectory)
+      loadPluginInfoFromDirectory(tempDirectory)
     } catch (e: DecompressorSizeLimitExceededException) {
       return PluginCreationFail(PluginFileSizeIsTooLarge(e.sizeLimit))
     } finally {
       tempDirectory.deleteLogged()
     }
   }
 
-  private fun loadDescriptorFromDirectory(pluginDirectory: File): PluginCreationResult<HubPlugin> {
+  private fun loadPluginInfoFromDirectory(pluginDirectory: File): PluginCreationResult<HubPlugin> {
     val errors = validateHubPluginDirectory(pluginDirectory)
     if (errors != null) {
       return errors
     }
-    val descriptorFile = File(pluginDirectory, DESCRIPTOR_NAME)
-    if (descriptorFile.exists()) {
-      return loadDescriptor(descriptorFile)
+    val manifestFile = File(pluginDirectory, DESCRIPTOR_NAME)
+    if (!manifestFile.exists()) {
+      return PluginCreationFail(PluginDescriptorIsNotFound(DESCRIPTOR_NAME))
     }
-    return PluginCreationFail(PluginDescriptorIsNotFound(DESCRIPTOR_NAME))
+    val manifestContent = manifestFile.readText()
+    val manifest = Json(
+        JsonConfiguration.Stable.copy(isLenient = true, ignoreUnknownKeys = true)
+    ).parse(HubPluginManifest.serializer(), manifestContent)
+    val icon = try {
+      loadIcon(pluginDirectory, manifest.iconUrl)
+    } catch (e: IllegalArgumentException) {
```

Comment:
bad code style (unchecked exceptions mustn't be caught), move this check to `loadIcon` method

Reply:
I can't move it to loadIcon, this catch clause is used to return error response if excpetion is thrown during icon loading